Fix type_traits for newer compilers (both, gcc-4.9.4 and gcc-6.2.0)
There is no std::tr1::has_trivial_constructor etc. anymore.

--- a/src/3rdparty/javascriptcore/JavaScriptCore/wtf/TypeTraits.h	2016-09-12 01:09:46.592885353 +0200
+++ b/src/3rdparty/javascriptcore/JavaScriptCore/wtf/TypeTraits.h	2016-09-12 01:20:07.594161571 +0200
@@ -166,7 +166,7 @@
         typedef T Type;
     };
 
-#if (defined(__GLIBCXX__) && (__GLIBCXX__ >= 20070724) && defined(__GXX_EXPERIMENTAL_CXX0X__)) || (defined(_MSC_VER) && (_MSC_VER >= 1600))
+#if (defined(__GLIBCXX__) && (__GLIBCXX__ >= 20070724) && (__GLIBCXX__ < 20160726) && defined(__GXX_EXPERIMENTAL_CXX0X__)) || (defined(_MSC_VER) && (_MSC_VER >= 1600))
 
     // GCC's libstdc++ 20070724 and later supports C++ TR1 type_traits in the std namespace.
     // VC10 (VS2010) and later support C++ TR1 type_traits in the std::tr1 namespace.
--- a/src/3rdparty/webkit/Source/JavaScriptCore/wtf/TypeTraits.h	2015-05-07 16:14:45.000000000 +0200
+++ b/src/3rdparty/webkit/Source/JavaScriptCore/wtf/TypeTraits.h	2016-09-12 01:36:22.347445737 +0200
@@ -179,7 +179,7 @@
         typedef T Type;
     };
 
-#if (defined(__GLIBCXX__) && (__GLIBCXX__ >= 20070724) && defined(__GXX_EXPERIMENTAL_CXX0X__)) || (defined(_MSC_VER) && (_MSC_VER >= 1600))
+#if (defined(__GLIBCXX__) && (__GLIBCXX__ >= 20070724) && (__GLIBCXX__ < 20160726) && defined(__GXX_EXPERIMENTAL_CXX0X__)) || (defined(_MSC_VER) && (_MSC_VER >= 1600))
 
     // GCC's libstdc++ 20070724 and later supports C++ TR1 type_traits in the std namespace.
     // VC10 (VS2010) and later support C++ TR1 type_traits in the std::tr1 namespace.
